src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/DefaultForegroundDispatcher.cs

```diff
@@ -5,24 +5,21 @@
 using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.CodeAnalysis.Razor;
 
-namespace Microsoft.AspNetCore.Razor.LanguageServer.Common
+namespace Microsoft.CodeAnalysis.Razor.Workspaces
 {
     internal class DefaultForegroundDispatcher : ForegroundDispatcher
     {
-        public override bool IsForegroundThread => Thread.CurrentThread.ManagedThreadId == ForegroundTaskScheduler.Instance.ForegroundThreadId;
+        public override bool IsForegroundThread => System.Environment.CurrentManagedThreadId == ForegroundTaskScheduler.Instance.ForegroundThreadId;
```

Comment:
I actually had no idea the `System.Environment.CurrentManagedThreadId` API existed. Is it any different than `Thread.CurrentThread.ManagedThreadID`? I lean towards using the latter only for consistencies sake but don't know enough

Reply:
I have no idea, tbh I changed it because it was actually an analyzer suggestion. 😅 looking at the docs I can't see any reason why `Thread.CurrentThread.ManagedThreadId` shouldn't be used though, so I can change it back.

Reply:
Ya I'd change it back just for consistency with the past / certainty it does what we think 😄

Reply:
Sounds good! I also filed https://github.com/dotnet/roslyn-analyzers/issues/5197 because I have no idea why the diagnostic was being offered, and the diagnostic info page gives an error